tests/functional/test_uninstall.py

```diff
@@ -59,6 +59,26 @@ def test_basic_uninstall_distutils(script):
     ) in result.stderr
 
 
+def test_uninstall_archive(script, caplog):
+    name = "GTrolls.tar.gz"
+    result = script.pip("uninstall", name, "-y", expect_error=True)
+    expected_message = (
+        f"Invalid requirement: '{name}' ignored -"
+        f" the uninstall command expects named requirements."
+    )
+    assert expected_message in result.stderr
+
+
+def test_uninstall_url(script, caplog):
```

Comment:
this test could(should?) be merged with the above one using a pytest.mark.parametrize .. ?